src/AppBundle/Controller/EventDataController.php

```diff
@@ -197,6 +197,9 @@ public function generateStatsAction(JobHandler $jobHandler): Response
             // JavaScript will poll for job status and update the view accordingly.
             $this->session->save();
 
+            // To process the job in this web request, bump execution time limit to 30 minutes.
+            set_time_limit(30 * 60);
```

Comment:
Wouldn't we want to do an hour, to match `Event::getStaleJobs()`? I think the time constraints should live as constants in `Job`. We don't actually pass in anything to getStaleJobs() where it is used, so you could get rid of that argument and have that method use Job::MAX_LIFE (or whatever you want to call it). You'd use that constant here, too.